go/client/chat_api_listen_display.go

```diff
@@ -98,6 +121,42 @@ func (d *chatNotificationDisplay) formatMessage(inMsg chat1.IncomingMessage) *Me
 	}
 }
 
+func matchMembersTypeToFilter(filter ChatChannel, typ chat1.ConversationMembersType) bool {
+	if filter.MembersType == "" {
+		// Default empty value matches to any non-team convo.
+		switch typ {
+		case chat1.ConversationMembersType_KBFS, chat1.ConversationMembersType_IMPTEAMNATIVE, chat1.ConversationMembersType_IMPTEAMUPGRADE:
+			return true
+		default:
+			return false
+		}
+	}
+	return filter.MembersType == typ.String()
+}
+
+func (d *chatNotificationDisplay) matchFilters(conv *chat1.InboxUIItem) bool {
+	if len(d.filtersNormalized) == 0 {
+		// No fliters - every message is relayed.
+		return true
+	}
+	for _, v := range d.filtersNormalized {
+		if matchMembersTypeToFilter(v, conv.MembersType) && v.Name == strings.ToLower(conv.Name) {
+			// If any of the following were specified by user and differ from
+			// what was received, filter the msg out.
+			if v.TopicType != "" && conv.TopicType.String() != v.TopicType {
```

Comment:
why not combine these checks into the conditional as well?

Reply:
so fold all checks into one ifs? or extract to separate function or to `matchMembersTypeToFilter` (probably renamed then)? thanks

Reply:
whichever you think makes it most readable, probably putting them all into one `if` and putting each condition on a new line is easiest